site/docs/starter/slash-commands.md

```diff
@@ -0,0 +1,550 @@
+  for (const command of commands.values()) {
+    if (command.scope) {
+      // Since we are only updating guild commands, we should only select commands with the "Guild" scope.
+      if (command.scope === "Guild") {
+        perGuildCommands.push({
+          name: command.name,
+          description: command.description,
+          type: command.type,
+          options: command.options ? command.options : undefined,
+        });
+      }
+    }
+  }
+
+  if (perGuildCommands.length) {
+    // Upsert will add commands, replace existing commands, and delete ones that do not exist anymore.
+    await upsertApplicationCommands(bot, perGuildCommands, guild.id);
+  }
+}
+// ...
+```
+
+We'll add this to our `mod.ts` to make sure that we always have the latest commands ready for users:
+
+```typescript title="mod.ts"
+// Imports
+import { logger } from "./src/utils/logger.ts";
+import {
+  dotEnvConfig,
+  enableCachePlugin, 
+  createBot, 
+  GatewayIntents, 
+  enableCacheSweepers, 
+  startBot 
+} from "./deps.ts";
+import { events } from "./src/events/mod.ts";
+// highlight-next-line
+import { updateCommands } from "./src/utils/helpers.ts";
+
+const log = logger({ name: "Main" });
+
+// Secrets
+dotEnvConfig({ export: true });
+export const BOT_TOKEN = Deno.env.get("BOT_TOKEN") || "";
+export const BOT_ID = BigInt(atob(BOT_TOKEN.split(".")[0]));
+
+log.info("Starting Bot, this might take a while...");
+
+// Create the Bot Object
+export const bot = enableCachePlugin(createBot({
+  token: BOT_TOKEN,
+  botId: BOT_ID,
+  intents: GatewayIntents.Guilds | GatewayIntents.GuildMembers | GatewayIntents.GuildMessages | GatewayIntents.DirectMessages,
+  events
+}));
+
+enableCacheSweepers(bot);
+
+if (BOT_TOKEN) {
+  // Start the bot
+  await startBot(bot);
+  // highlight-start
+  // Update all commands
+  await updateCommands(bot);
+  // highlight-end
+} else {
+  throw "Error: No Token!";
+}
+```
+
+Then add the following to the `guildCreate` event handler to make sure new guilds also get our guild only commands:
+
+## Handling Commands
+
+We're nearly there! Our bot can now define and register commands; which is great, but it can't actually run any
+commands. If we were to start the bot and run `/ping` in discord, it would show an error message and we would only log
+that the interaction happened (see step 4). Let's fix that by adding onto the `interactionCreate` event handler:
+
+```typescript title="src/events/interactionCreate.ts"
+import {
+  ApplicationCommandOptionTypes,
+  bgBlack,
+  bgYellow,
+  black,
+  BotWithCache,
+  green,
+  Guild,
+  red,
+  white,
+  yellow,
+} from "../../deps.ts";
+import { Command, commands } from "../commands/mod.ts";
+import { getGuildFromId, isSubCommand, isSubCommandGroup } from "../utils/helpers.ts";
+import { logger } from "../utils/logger.ts";
+import { events } from "./mod.ts";
+
+const log = logger({ name: "Event: interactionCreate" });
+
+events.interactionCreate = async (rawBot, interaction) => {
```

Comment:
This is quite complicated for a starter guide, what if this was a plugin so that you could do `events.command` `events.button` or `events.modal` etc... similar to events.messageCreate

Reply:
Like in a DD plugin?